GitCommands/UserRepositoryHistory/RepositoryXmlSerialiser.cs

```diff
@@ -0,0 +1,118 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Xml.Serialization;
+using JetBrains.Annotations;
+
+namespace GitCommands.UserRepositoryHistory
+{
+    /// <summary>
+    /// Provides the ability to serialise and deserialise collections of user's git repositories.
+    /// </summary>
+    public interface IRepositorySerialiser
+    {
+        /// <summary>
+        /// Restores a list of user's git repositories from the supplied string.
+        /// </summary>
+        /// <param name="serialised">A serialised list of user's git repositories.</param>
+        /// <returns>A list of user's git repositories.</returns>
+        IReadOnlyList<Repository> Deserialize(string serialised);
+
+        /// <summary>
+        /// Serialises the given list of user's git repositories.
+        /// </summary>
+        /// <param name="repositories">A list of user's git repositories.</param>
+        /// <returns>A serialised list of user's git repositories.</returns>
+        string Serialize(IEnumerable<Repository> repositories);
+    }
+
+    /// <summary>
+    /// Serialises or deserialises user's git repositories into XML.
+    /// </summary>
+    public sealed class RepositoryXmlSerialiser : IRepositorySerialiser
+    {
+        /// <summary>
+        /// Restores a list of user's git repositories from the supplied string.
+        /// </summary>
+        /// <param name="serialised">A serialised list of user's git repositories.</param>
+        /// <returns>A list of user's git repositories, if successfully deserialised; otherwise <see langword="null"/>.</returns>
+        /// <exception cref="ArgumentException"><paramref name="serialised"/> is <see langword="null"/> or <see cref="string.Empty"/>.</exception>
+        [ContractAnnotation("serialised:null=>halt")]
+        public IReadOnlyList<Repository> Deserialize([NotNull]string serialised)
+        {
+            if (string.IsNullOrEmpty(serialised))
+            {
+                throw new ArgumentException(nameof(serialised));
+            }
+
+            try
+            {
+                var serializer = new XmlSerializer(typeof(RepositoryHistorySurrogate));
+                using (TextReader reader = new StringReader(serialised))
+                {
+                    if (serializer.Deserialize(reader) is RepositoryHistorySurrogate obj)
+                    {
+                        return obj.Repositories;
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Trace.WriteLine(ex.Message);
```

Comment:
❓ Is this a standard approach already used in the code base? If yes, then we can leave any changes for a future PR.

Reply:
To be totally honest I can't tell. I have seen this in few places but I don't think the codebase has one consistent approach dealing with exceptions.
This is the existing code and it was outside the scope of the this refactor.
This is one of a million refactors to we need to do.